### DISCUSSION TITLE
convert_names_case  and  snake_case_fallback_resolvers

### DISCUSSION THREAD

**Discussion by chris-fibo** [mirumee/ariadne#1102] [Q&A]
The claim is  snake_case_fallback_resolvers is being deprecated.
I removed the resolve and set convert_names_case  to true in my make_executable_schema call.
With snake_case_fallback_resolvers set, everything works.
When I remove it and set convert_names_case  I get an error saying one of my required fields is null (This is as far as my testing has gone, so there may be other issues.

```
GraphQl:
type User {
  isUserActive: Boolean!
  ...
  }
```
  
My object has a "is_user_active" field. When I return this object I get the error that isUserActive isn't set. 

Now when I use convert_names_case, it appears to work as a replacement for @convert_kwargs_to_snake_case (although I haven't completed my testing, I did see another discussion that nest objects don't seem to be converted)
But is convert_names_case=True also replacing snake_case_fallback_resolvers?

**Comment by rafalp** [comment]
Can you show minimal reproduction code for the issue? We have tests in place for this and they work.

**Comment by mdp-chris** [comment]
This got put on a back burner, and I switched jobs. But I figured out what was going on.
I was following the example for creating a directive. 
```
class DateDirective(SchemaDirectiveVisitor):
    def visit_field_definition(self, field, object_type):
        original_resolver = field.resolve or default_field_resolver

        def resolve_formatted_date(obj, info, **kwargs):
            ...
            return result.isoformat()

        field.resolve = resolve_formatted_date
        return field
```

I think this code is executed early on, and adds a resolver to anything with the directive.
When you set convert_names_case=True then at start time, it will run through and essentially add a special resolver to any field that doesn't have a resolver. But the above code adds the default_field_resolver, and it doesn't resolve to the camel case. 
I think the best solution would be for ariadne to offer its own default_field_resolver or check to see if the existing resolver is the default resolver and if it is treat it like it is missing, but this is how I solved it:

```
  from ariadne.resolvers import resolve_to
  from ariadne.utils import convert_camel_case_to_snake

      def visit_field_definition(self, field: GraphQLField, _) -> GraphQLField:

        def camel_case_resolver(obj, info, **kwargs):
            resolve = resolve_to(convert_camel_case_to_snake(info.field_name))
            return resolve(obj, info, **kwargs)

        original_resolver = field.resolve or camel_case_resolver

        def resolve_is_authorized(obj, info, **kwargs):
          ...
```

Instead of using the default resolver, I just do the same that that ariadne is doing if the resolver was missing.
